Video Poker - RA_files/vpoker-init-script.js

```diff
@@ -0,0 +1,408 @@
+  // if (resetState === true && canDeal === true){
+  //   // points = 100;
+  //   // resetState = false;
+  //   // canDeal = false;
+  //   location.reload()
+  // }
+};
+
+//bet amount callback events - using existing points as credit
+const increaseBet = () => {
+  if (betAmount >= 1 && canDeal === true && points >= 2){
+    betAmount += 1
+    points -= 1
+    console.log(`new bet amount is ${betAmount}`)
+    currentBet.innerText = `Amount Bet: ${betAmount}`;
+    pointsInfo.innerText = `Points: ${points}`;
+
+  }
+}
+
+const decreaseBet = () => {
+  if (betAmount > 1 && canDeal === true){
+    betAmount -= 1
+    points += 1
+    console.log(`new bet amount is ${betAmount}`)
+    currentBet.innerText = `Amount Bet: ${betAmount}`;
+    pointsInfo.innerText = `Points: ${points}`;
+  }
+}
+
+
+
+/* #######################
+## GAME INITIALISATION ##
+####################### */
+const initGame = () => {
+  cardContainer = document.createElement('div');
+
+  // fill game info div with starting instructions
+  gameInfo.innerText = `Place your Bet Amount, then Draw 5 Cards!
+  
+  `;
+  
+  // display points up top
+  pointsInfo.innerText = `Points: ${points}`;
+  document.body.appendChild(pointsInfo);
+
+  // create row1, append to main card container
+  const row1 = document.createElement('div');
+  row1.setAttribute('class', 'row');
+  // row1.innerText = "Player 1's Cards";
+  cardContainer.appendChild(row1);
+
+  // create card container for player 1, append to row 1
+  cardContainerPlayer1 = document.createElement('div');
+  cardContainerPlayer1.classList.add('card-container-player-1');
+  row1.appendChild(cardContainerPlayer1);
+
+  // append all the containers, rows and sub-containers to the document
+  document.body.appendChild(cardContainer);
+
+  //display game info just below the hand dealt
+  document.body.appendChild(gameInfo);
+
+  // build a row for the only buttons used in-game
+  const row2 = document.createElement('div');
+  row2.setAttribute('class', 'button-row');
+
+  // initialize deal button functionality and append with listener
+  player1Button.innerText = 'Deal Cards';
+  player1Button.setAttribute('class', 'button');
+  player1Button.addEventListener('click', player1Click);
+  row2.appendChild(player1Button);
+
+  // initialize betting button functionality and append with listener
+  const betButtons = document.createElement('div')
+  
+  betUp.innerText = '+';
+  betUp.setAttribute('class', 'button');
+  betUp.addEventListener('click', increaseBet);
+
+  betDown.setAttribute('class', 'button');
+  betDown.innerText = '-';
+  betDown.addEventListener('click', decreaseBet);
+
+  currentBet.setAttribute('id', 'bet-amount');
+  currentBet.innerText = `Amount Bet: ${betAmount}`;
+  
+  betButtons.appendChild(betUp);
+  betButtons.appendChild(currentBet);
+  betButtons.appendChild(betDown);
+
+  row2.appendChild(betButtons);
+  document.body.appendChild(row2);
+};
```

Comment:
consider putting this block of code in a helper function so that it is easier to read